From 2f7d2d02e71516bcc09f691f1c4767b08b914e4f Mon Sep 17 00:00:00 2001
From: Kenneth Kasilag <kenneth@kasilag.me>
Date: Wed, 18 Mar 2026 03:23:21 +0000
Subject: [PATCH] (pr-22473) airoha: an7581: enable ramoops support

Enable ramoops to store crash data or pass boot settings
to bootloader on soft-reboot.

Signed-off-by: Kenneth Kasilag <kenneth@kasilag.me>
---
 target/linux/airoha/an7581/config-6.18 | 5 +++++
 target/linux/airoha/dts/an7581.dtsi    | 8 ++++++++
 2 files changed, 13 insertions(+)

diff --git a/target/linux/airoha/an7581/config-6.18 b/target/linux/airoha/an7581/config-6.18
index 8ccea941f30467..35399c2a8425c2 100644
--- a/target/linux/airoha/an7581/config-6.18
+++ b/target/linux/airoha/an7581/config-6.18
@@ -311,6 +311,11 @@ CONFIG_POSIX_CPU_TIMERS_TASK_WORK=y
 CONFIG_POWER_RESET=y
 CONFIG_POWER_RESET_SYSCON=y
 CONFIG_POWER_SUPPLY=y
+CONFIG_PSTORE=y
+CONFIG_PSTORE_COMPRESS=y
+CONFIG_PSTORE_CONSOLE=y
+CONFIG_PSTORE_PMSG=y
+CONFIG_PSTORE_RAM=y
 CONFIG_PTP_1588_CLOCK_OPTIONAL=y
 CONFIG_QUEUED_RWLOCKS=y
 CONFIG_QUEUED_SPINLOCKS=y
diff --git a/target/linux/airoha/dts/an7581.dtsi b/target/linux/airoha/dts/an7581.dtsi
index 1db83dbb9796e6..f9c1abea514786 100644
--- a/target/linux/airoha/dts/an7581.dtsi
+++ b/target/linux/airoha/dts/an7581.dtsi
@@ -30,6 +30,14 @@
 			reg = <0x0 0x84000000 0x0 0xa00000>;
 		};
 
+		/* 64 KiB reserved for ramoops/pstore */
+		ramoops@86ff0000 {
+			no-map;
+			compatible = "ramoops";
+			reg = <0x0 0x86ff0000 0x0 0x10000>;
+			record-size = <0x1000>;
+		};
+
 		qdma0_buf: qdma0-buf@87000000 {
 			no-map;
 			reg = <0x0 0x87000000 0x0 0x2000000>;
